daml-foundations/daml-ghc/tests/ConsumingTests.daml

```diff
@@ -0,0 +1,43 @@
+daml 1.2
+module ConsumingTests where
+
+template T
+  with
+    p : Party
+  where
+    signatory p
+
+    nonconsuming choice NoConsume : Int
+      controller p
+      do
+        return 41
+
+    choice PreConsume : Int
+      controller p
+      do
+        return 42
+
+-- Temporary : Will replace with DAML syntax when it's implemented.
+
+data PostConsume = PostConsume {} deriving (Eq, Show)
+instance Choice T PostConsume Int where
+  consuming = postconsuming
+  choiceController (T{..}) _ = [p]
+  choice _ self _ =  do _ <- fetch self; return 43 -- Can fetch 'self' in choice body
```

Comment:
Can we write this as `(+2) <$> exercise self Nonconsume` as that's what our user actually wants. Perhaps add a fetch too to check that as well.

Reply:
✔️